myhpom/views/document.py

```diff
@@ -0,0 +1,35 @@
+import mimetypes
+import requests
+from ipware import get_client_ip
+from django.http import Http404, HttpResponse, FileResponse
+from django.utils.timezone import now
+from django.contrib.sites.models import Site
+from myhpom.models import DocumentUrl
+
+
+def document_url(request, key, filename):
+    """download the document at the given url.
+    * key: the DocumentUrl.key by which to find the document
+    * filename: included in the URL to provide a sensible filename and to verify the URL
+    """
+    # protect on client_ip, doc_url match, expiration, and filename
+    client_ip, _ = get_client_ip(request)
+    doc_url = DocumentUrl.objects.filter(key=key).first()
```

Comment:
This could be simplified to `DocumentUrl.objects.get()` within a try block for the DoesNotExist exception.